tiny_dnn/parameter.h

```diff
@@ -0,0 +1,114 @@
+/*
+    Copyright (c) 2013, Taiga Nomi and the respective contributors
+    All rights reserved.
+
+    Use of this source code is governed by a BSD-style license that can be found
+    in the LICENSE file.
+*/
+#pragma once
+
+#include "tiny_dnn/core/framework/tensor.h"
+#include "tiny_dnn/util/util.h"
+
+namespace tiny_dnn {
+
+enum class parameter_type : int8_t { weight = 0x1, bias = 0x2 };
```

Comment:
I'm not really sure why this enum has to be 1 byte object...
Is it really meaningful to save storage size of `parameter::type_`?